packages/theme/src/cli/services/dev.ts

```diff
@@ -98,47 +107,93 @@ export async function dev(options: DevOptions) {
   await renderDevSetupProgress()
   await serverStart()
 
-  renderLinks(options.store, String(options.theme.id), host, port)
+  renderLinks(urls)
   if (options.open) {
-    openURL(`http://${host}:${port}`).catch((error: Error) => {
-      renderWarning({headline: 'Failed to open the development server.', body: error.stack ?? error.message})
+    openURLSafely(urls.local, 'development server')
+  }
+
+  readline.emitKeypressEvents(process.stdin)
+  if (process.stdin.isTTY) {
+    process.stdin.setRawMode(true)
+  }
+
+  process.stdin.on('keypress', (_str, key) => {
+    if (key.ctrl && key.name === 'c') {
+      process.exit()
+    }
+
+    switch (key.name) {
+      case 't':
+        openURLSafely(urls.local, 'localhost')
+        break
+      case 'p':
+        openURLSafely(urls.preview, 'theme preview')
+        break
+      case 'e':
+        openURLSafely(urls.themeEditor, 'theme editor')
+        break
+      case 'g':
+        openURLSafely(urls.giftCard, 'gift card preview')
+        break
+    }
+  })
+}
+
+export function openURLSafely(url: string, label: string) {
+  openURL(url).catch(handleOpenURLError(label))
+}
+
+function handleOpenURLError(message: string) {
+  return (error: Error) => {
+    renderWarning({
+      headline: `Failed to open ${message}.`,
+      body: error.stack ?? error.message,
     })
   }
 }
 
-export function renderLinks(store: string, themeId: string, host = DEFAULT_HOST, port = DEFAULT_PORT) {
-  const remoteUrl = `https://${store}`
-  const localUrl = `http://${host}:${port}`
+export function renderLinks(urls: {local: string; giftCard: string; themeEditor: string; preview: string}) {
   renderSuccess({
     body: [
       {
         list: {
-          title: {bold: 'Preview your theme'},
-          items: [localUrl],
+          title: chalk.bold('Preview your theme ') + chalk.cyan('(t)'),
```

Comment:
I believe we could follow the CLI-kit standards to present the shortcuts. Thus, this line could be presented as:

```ts
title: [{bold: 'Preview your theme'}, {subdued: '(t)'}],
```

or:
```ts
title: [{bold: 'Preview your theme'}, {command: '(t)'}],
```

I believe that `subdued` fits a bit more in this context. While it's a bit tempting to be opinionated about the color, it's nice to use CLI-kit standard colors, so we may make the CLI more cohesive across commands :)

Reply:
After doing some digging, this won't work for labels as they expect a string. We will have to make some changes in cli-kit. I have made a follow up issue [here](https://github.com/Shopify/develop-advanced-edits/issues/438) to make that fix.